### PR TITLE
Contact helper issues

### LINKED ISSUE #444: ENH: contact updater can take an id as an optional argument
When updating (rather than adding) it would be nice to be able to specify an id rather than first/last name only.

Come to think of it, it would be nice to be able to specify an id in any case, for example, with chinese names (and possibly other nationalities in the future) our "standard" behavior is to put two initials from the first name (so Songsheng Tao is sstao, not stao)

### PR DESCRIPTION
**Pull request by dy2403**
closes #444 and #443 

hi @sbillinge !
can you please review it?
I am trying to figure out why the error is happening with your empty list (issue 445). 
the problem seems to be linked to current, and not to notes though.

### CODE REVIEW

File: regolith/helpers/u_contacthelper.py [hunk 1]
```diff
@@ -16,9 +16,12 @@
 
 def subparser(subpi):
     subpi.add_argument("name", help="first name space last name in quotes")
-    subpi.add_argument("institution", help="person's institution.  Can be inf"
-                                            "short form such as columbiau and will "
-                                            "be retrieved from institutions collection")
+    subpi.add_argument("--id", help="id of the person, e.g., first letter first name "
+                                    "plus last name, but unique")
+    subpi.add_argument("--inst", help="person's institution.  Can be inf "
```
Comment (sbillinge) [inline regolith/helpers/u_contacthelper.py:21]: maybe have a single letter abbreviation, ``-i`` for institution and ``-d` for id?

Also inf is a typo?

File: regolith/helpers/u_contacthelper.py [hunk 2]
```diff
@@ -82,7 +88,11 @@ def db_updater(self):
             notes = current.get('notes', [])
             aliases = current.get('aka', [])
         else:
-            pdoc.update({"date": dt.today()})
+            if not rc.inst:
```
Comment (sbillinge) [inline regolith/helpers/u_contacthelper.py:91]: I think you need to raise an exception here, not just a print statement

File: regolith/helpers/u_contacthelper.py [hunk 3]
```diff
@@ -82,7 +88,11 @@ def db_updater(self):
...
+                print("Institution (--inst) is required to create a new contact")
+                return
+            pdoc.update({"date": dt.date.today()})
+            pdoc.update({"institution": rc.inst})
             notes = []
             aliases = []
             UUID = str(uuid.uuid4())
```
Comment (sbillinge) [inline regolith/helpers/u_contacthelper.py:98]: this should be lower-case for a string variable.

Comment (dy2403) [reply regolith/helpers/u_contacthelper.py]: changed to uniqueidentifier, it gets angry if I try to use uuid

File: regolith/helpers/u_contacthelper.py (hunk 2 above)
Comment (dy2403) [reply regolith/helpers/u_contacthelper.py]: changed to:
`            if not rc.institution:
                raise RuntimeError("Institution is required to create a new contact")`

File: regolith/helpers/u_contacthelper.py (hunk 1 above)
Comment (dy2403) [reply regolith/helpers/u_contacthelper.py]: done!

**Comment by dy2403** [comment]
@sbillinge yes, can you review it and merge? 
I wasn't sure about the news, hope I did it correctly

**Comment by sbillinge** [comment]
thanks @dy2403

**Pull request merged**